pkg/orchestrator/orchestrator.go

```diff
@@ -107,7 +107,7 @@ type ConnectorService interface {
 type ProcessorService interface {
 	List(ctx context.Context) map[string]*processor.Instance
 	Get(ctx context.Context, id string) (*processor.Instance, error)
-	Create(ctx context.Context, id string, procType string, parent processor.Parent, cfg processor.Config, p processor.ProvisionType) (*processor.Instance, error)
+	Create(ctx context.Context, id string, procType string, parent processor.Parent, cfg processor.Config, p processor.ProvisionType, condition string) (*processor.Instance, error)
```

Comment:
Nitpick: would be good to document what's expected to be in the string (a Go template)?

Alternatively, we could maybe have `type GoTemplate string`.

Reply:
i prefer adding docs, IMO adding more types will make it more complicated going forward with casting and type changing, I think "condition" is similar to the "type" field, which is also just a string